state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1224,3 +1234,16 @@ pub mod pallet {
 		}
 	}
 }
+
+impl<T: Config> SwapQueueApi for Pallet<T> {
+	type BlockNumber = BlockNumberFor<T>;
+
+	fn schedule_swap(
+		from: Asset,
+		to: Asset,
+		amount: AssetAmount,
+		swap_type: SwapType,
+	) -> (u64, Self::BlockNumber) {
+		Self::schedule_swap_internal(from, to, amount, swap_type)
```

Comment:
I don't think we want to do the MaximumSwapAmount checks for a network fee swap

Reply:
if we add it to the SwapQueue and use `Self::swap_into_stable_taking_network_fee(&mut swaps)?;` which is what's happening now, then we'll be taking a network fee for the network fee swap. which we don't want either. 
We should add a test for this case - after the network swap goes through then we should have nothing left to swap in the Network fee storage

Reply:
I'm wondering if it makes sense to have a second queue here. What matters isn't that it's in the `SwapQueue` specifically - but that it is queued by the delay and it shows in the subscription. I think this is probably the cleaner solution than trying to retrofit this into the existing flow which takes network fees etc. Also means we get the ability to emit a separate event for the network swap (if we wanted to). 

CC: @dandanlen ^

Reply:
I don't think we really care that much if the network fee swap also pays a fee. Adding another queue raises more questions than it answers (do bundle it with other swaps? why/why not? what about delays, slippage limits etc...) 

Much simpler to treat it like a normal swap - notify the LPs, bundle it with everything else, recover like everything else.

Reply:
Yeah I guess it's not a big deal

Reply:
If we really care about this, we can address it in the swap bundling / unbundling logic. Fee Swaps are always USDC->FLIP, so they happen on the second leg. We could just add the network fee swap to the bundle after we take the network fee. 

This would be less invasive than adding a whole new queue.

Reply:
Is it not as simple as just checking swap_type in `swap_into_stable_taking_network_fee` to see if the fee should be taken? Something like:

```rust
fn swap_into_stable_taking_network_fee(
	swaps: &mut Vec<Swap>,
) -> Result<(), BatchExecutionError> {
	Self::do_group_and_swap(swaps, SwapLeg::ToStable)?;

	// Take NetworkFee for all swaps
	for swap in swaps.iter_mut() {
                 // ***** NEW CODE BEGINS *****
		if swap.swap_type == SwapType::NetworkFee {
			// Don't take network fee for network fee swaps
			continue;
		}
                 // ***** NEW CODE END *****
		debug_assert!(
			swap.stable_amount.is_some(),
			"All swaps should have Stable amount set here"
		);
		let stable_amount = swap.stable_amount.get_or_insert_with(Default::default);
		*stable_amount = T::SwappingApi::take_network_fee(*stable_amount);
	}

	Ok(())
}
```